tyk-headless/templates/deployment-gw-repset.yaml

```
@@ -48,6 +48,14 @@ spec:
       affinity:
 {{ toYaml .Values.gateway.affinity | indent 8 }}
 {{- end }}
+      initContainers: 
+      - name: "setup-directories"
+        image: busybox
+        command: ['sh','-c','mkdir -p apps && mkdir -p middlewares && mkdir -p policies']
```

This can be simplified:
```suggestion
        command: ['sh','-c','mkdir -p apps middleware policies']
```

Also seems like the config expects `/mnt/tyk-gateway/middleware` (not `middlewares`):
https://github.com/TykTechnologies/tyk-helm-chart/blob/fdff092f17291f16e4f3823c256ac2c91418e8c3/tyk-headless/configs/tyk_mgmt.conf#L8